commit a89ca8d356aed48694d1024a910a273015d24863
Author: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date:   Wed Nov 13 02:39:14 2013 -0500

    deal with XPCOM_ABI names with slash (like "GNU/kFreeBSD")
    
    Without this patch, the sed expressions in genxpi get tripped up when
    they encounter an XPCOM_ABI with a slash in it.

diff --git a/genxpi b/genxpi
index d67ee3a..16e00da 100755
--- a/genxpi
+++ b/genxpi
@@ -63,8 +63,9 @@ if [ "$xpcomAbi" = "" ] ; then
   xpcomAbi="unknown"
 fi
 platform=${osArch}_${xpcomAbi}
+escplatform="$(printf "%s" "${platform}" | sed 's_/_\\/_g')"
 # Pepare install.rdf
-sed 's/<!-- targetPlatform placeholder -->/<em:targetPlatform>'${platform}'<\/em:targetPlatform>/' < ${srcDir}/package/install.rdf > ${targetDir}/install.rdf
+sed 's/<!-- targetPlatform placeholder -->/<em:targetPlatform>'${escplatform}'<\/em:targetPlatform>/' < ${srcDir}/package/install.rdf > ${targetDir}/install.rdf
 
 
 enigmimeDll=${libPrefix}enigmime${dllSuffix}
@@ -82,7 +83,7 @@ spDllFile=platform/${platform}/lib/${libPrefix}subprocess-${xpcomAbi}${dllSuffix
 touch ${targetDir}/chrome.manifest
 cp  ${targetDir}/chrome.manifest ${targetDir}/chrome.manifest.save
 cat  ${srcDir}/package/chrome.manifest | \
-sed  's/##ENIGMIMEDLL-PLACEHOLDER##/binary-component platform\/'${platform}'\/components\/'`basename ${enigDllFile}`' ABI='${platform}'/' \
+sed  's/##ENIGMIMEDLL-PLACEHOLDER##/binary-component platform\/'${escplatform}'\/components\/'`basename ${enigDllFile}`' ABI='${escplatform}'/' \
 > ${targetDir}/chrome.manifest
 
 # Prepare languages other than en-US
